x-pack/plugins/apm/public/components/shared/redirect_with_default_date_range/index.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ReactElement } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { useDateRangeRedirect } from '../../../hooks/use_date_range_redirect';
+
+export function RedirectWithDefaultDateRange({
+  children,
+}: {
+  children: ReactElement;
+}) {
+  const { isDateRangeSet, redirect } = useDateRangeRedirect();
+
+  const apmRouter = useApmRouter();
+  const location = useLocation();
+
+  const matchingRoutes = apmRouter.getRoutesToMatch(location.pathname);
+
+  if (
+    !isDateRangeSet &&
+    matchingRoutes.some((route) => {
+      return (
+        route.path === '/services' ||
+        route.path === '/traces' ||
+        route.path === '/service-map' ||
+        route.path === '/backends' ||
+        route.path === '/services/{serviceName}' ||
+        location.pathname === '/' ||
+        location.pathname === ''
+      );
```

Comment:
This isn't great. Ideally this would be a property of the route, as the route itself is disconnected from setting the default parameters. I've considered some alternatives, but they come at a cost. One issue is that the date range defaults are (AFAIK) only available via a hook, so we have to render a React element to get the appropriate defaults. We can't easily use a middleware layer for that reason (e.g. `onRouteActivate`). One option would be to allow the route to specify a `pre` element that gets rendered as an immediate child of `RouterProvider`, before any other elements, that can block rendering of child routes. That's a little bit of a weird API though, and I think we should wait for more use cases before implementing it.